pkg/ccl/changefeedccl/kvfeed/kv_feed.go

```diff
@@ -354,6 +369,17 @@ func (f *kvFeed) scanIfShould(
 		return err
 	}
 
+	// We will scan spansToBackfill up to and including scanTime.  That means that when we start
+	// rangefeed for those spans, we should start from scanTime.Next().
+	// Note: we forward frontier before we filter checkpoint spans.  This is safe since
+	// any error returned from this function is propagated to the caller, causing frontier
+	// to no longer be used.
+	for _, sp := range spansToBackfill {
+		if _, err := frontier.Forward(sp, scanTime.Next()); err != nil {
```

Comment:
It sounds strange to me to say "set the frontier after the scan time", since given the definition of the frontier that reads to me like "we're now saying that we've seen all events for this span up to **and including** right _after_ the scan time".   What causes us to need this?  Is it possible to instead have a `.Next()` further down at the point we're using this frontier information?  Even if it'd now be usable as a rangefeed start time, my concern is that at some point we do/would rely on the classic definition of frontier as "we've already seen this timestamp" and have some off-by-one issue.

Reply:
I think there is a bit of a misunderstanding.  When we scan spans, we scan up to and including that timestamp.
Therefore, when we start rangefeed, the feed for those spans should start from the next timestamp.\
That was the nature of the off-by 1 bug fix.  Now, the reason I did this manipulation in this function because
the set of spans and the timestamp depends on whether or not it's an initial scan and if there are table events.
At any rate, I changed the signature to return the list of spans that were scanned and their timestamp so that
we can advance frontier at the call site. Do you think this is more clear now?